### ISSUE TITLE
Live reload flag using outdated config.xml which is updated via Ionic hooks

### ISSUE THREAD

**Issue by sagrawal31** [ionic-team/ionic-cli#4579]
**Description:**
When we run a command like `ionic cordova run android --livereload`, it's picking the old (or very first) `config.xml` file which will be updated by Ionic `build:before` or `server:before` hooks.

**Steps to Reproduce:**

1. Create a fresh Ionic & Cordova app with a clean Git repository where the bundle id in `config.xml` is `com.example.foo`.
2. Define an Ionic hook, like- 
![image](https://user-images.githubusercontent.com/1804514/95341799-4d6cd880-08d4-11eb-949d-34936fb11d09.png)
3. The hook can simply update the `config.xml` bundle id to something like `com.example.bar`
4. Run the command- `ionic cordova run android --livereload`
5. Go & check that `<widget id=""` is again changed to `com.example.foo` and the Android app is built for `com.example.foo` instead of `com.example.bar`.

**My `ionic info`:**

```
Ionic:

   Ionic CLI                     : 6.11.6 (/usr/local/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 5.3.1
   @angular-devkit/build-angular : 0.1000.8
   @angular-devkit/schematics    : 9.1.12
   @angular/cli                  : 10.0.8
   @ionic/angular-toolkit        : 2.3.0

Cordova:

   Cordova CLI       : 10.0.0
   Cordova Platforms : android 9.0.0
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 5.0.0, (and 29 other plugins)

Utility:

   cordova-res                          : 0.15.1
   native-run (update available: 1.2.1) : 0.3.0

System:

   Android SDK Tools : 26.1.1 (/Users/shashank/Library/Android/sdk)
   ios-deploy        : 1.10.0
   ios-sim           : 8.0.2
   NodeJS            : v12.18.3 (/usr/local/bin/node)
   npm               : 6.14.6
   OS                : macOS Catalina
   Xcode             : Xcode 12.0 Build version 12A7209
```

**Other Information:**

This is happening because the `--livereload` flag update the `config.xml` for `<allow-navigation href="http://localhost:8100" sessionid="9b89ebf1" />` so it might be picking the content of `config.xml` before firing the hooks.

**Comment by imhoffd** [comment]
Good catch. This should be fixed in 6.11.12. Thanks!

**Comment by sagrawal31** [comment]
Thanks for very quick action @dwieeb Ill checkout the change and test it.
